### ISSUE TITLE
cumulative reward setting no longer being followed.

### ISSUE THREAD

**Issue by mrcoffee1026** [teozfrank/UltimateVotes#19]
You fixed the mtop display, but in doing so broke this setting:

```
    cumulativerewards:
      rewardbymonthlyvotescount: false

```
This setting is needed to prevent cumulative rewards from being given out for monthly counts instead of cumulative count.  Please fix asap.

**Comment by mrcoffee1026** [comment]
reverting to previous build, i'll be fixing ranks for at least a week now, guh...

**Comment by mrcoffee1026** [comment]
This is worth a bit of cash for the trouble if you'd be willing to fix this soon (and put a donate link somewhere).  Anyway I still need this, and the last time you did this you just basically broke the %votecount% placeholder by having it switch to the total count.  I still think there should be 2 placeholders, a %votecount% and a %monthlyvotecount%... and then I'd need to update my old messages.yml accordingly so that when a person does /vote mtop it shows the monthly instead of total count... without altering the way rewards are distributed.

**Comment by teozfrank** [comment]
Can you join my discord server so I can work with you better to resolve this quicker -> https://discord.gg/hn8e9tu

**Comment by mrcoffee1026** [comment]
you bet, thanks

**Comment by mrcoffee1026** [comment]
I've joined your discord but got busy doing something else and you're offline now.  To clarify the issue - if that's possible:
I do feel there should legitimately be two placeholders so that this issue doesn't come up.  You made an option to deal with half of this issue:
    cumulativerewards:
      rewardbymonthlyvotescount: false
This setting is now being ignored on the current build as you're checking the same placeholder, just adjusting the value which is returned for it based on this setting.  A monthly count and a total count should not be treated as the "same thing" based on any given setting for this very reason.  I award players rank when their total count reaches particular milestones.  If you arbitrarily switch to monthly totals for rewards, this basically randomizes everyone's ranks, promoting people before they should be, or skipping promotions they should have received.  I've also noticed that claiming offline votes sporadically either skips all rewards or rewards for monthly count... or rewards for both monthly and total in the same set of claims which also makes keeping track of ranks nearly impossible.  In any case the language of the message files tells you that your "monthly" count is as follows... but then uses the same placeholder that - given the setting that you added here to reward on total count - is also the total count... so when a person does /mtop they see their total count even though it says it is giving them their monthly count.  Trying to make this placeholder work in every situation by adding if statements for additional cases of option settings seems very prone to error and also... a strange approach since an entire database table is available for each value independently.  As they're stored as different numbers, why is the rest of the coding treating them like they're identical numbers?

I cannot come up with a way of logic or any method of reverse engineering what is occurring to rewards in an offline claim situation.  It appears that no rewards are being applied.  The addtestvote command will produce the worst of all possible worlds:  no awards given during original vote but... rewards will be given when 1) The monthly count reaches a cumulative reward number.  2) The total count reaches a cumulative reward number.  3) The number of claimed votes reaches a monthly reward number.... enabling my test user to skip 2-3 ranks in by claiming one batch of six votes.  Real-world live use of this just generally does not apply any rewards in most claiming situations, but when it does, seems to be using either the monthly count or the count of how many claimed votes had just been clicked... in either case it's not rewarding the cumulative rewards when they'd be expected.  There've been some rare times when this seems to be working... but then another player will claim some offline votes and... nope, sure isn't.

Thanks for your time, I'll catch up with you in discord when you're around.
